distribution/lib/Standard/Base/0.0.0-dev/src/Errors/Common.enso

```diff
@@ -380,3 +381,20 @@ type Dry_Run_Operation
        Convert the Dry_Run_Operation to a human-readable format.
     to_display_text : Text
     to_display_text self = self.message
+
+type Out_Of_Memory
+    ## PRIVATE
+       An error when Enso run's out of memory
+    Error operation
+
+    ## PRIVATE
+       Convert the Out_Of_Memory to a human-readable format.
+    to_display_text : Text
+    to_display_text self = "Enso ran out of memory performing the "+self.operation+" operation."
+
+    ## PRIVATE
+       ADVANCED
+       Catches possible `OutOfMemoryError` and throws an `Out_Of_Memory` error.
+    handle_java_exception operation ~function =
+        Panic.catch OutOfMemoryError handler=(_-> Error.throw (Out_Of_Memory.Error operation)) <|
```

Comment:
Isn't it generally frowned-upon to try to catch OOM?

Reply:
Catching in a few limited places where the user can easily create a very large table and cause the program to crash.

Reply:
Shouldn't we rather try to check if the allocation is safe _before_ performing it? (i.e. by checking available runtime memory)

Indeed it seems rather unsafe to catch OOME: https://stackoverflow.com/a/8729011 - the JVM can be left in an inconsistent state

Reply:
Happy to add a ticket to revisit all the allocations within Table operations but that is a massive task and we should probably do as part of the restructuring work.